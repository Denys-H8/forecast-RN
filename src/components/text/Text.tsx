import React from 'react';

import { Text, TextProps } from 'react-native';
import { styles } from './text.styles';

type Props = {
  isTitle?: boolean;
} & TextProps;

const CustomText: React.FC<Props> = ({ isTitle, style, children }) => {
  const textStyle = isTitle ? styles.title : styles.p;

  return <Text style={[textStyle, style]}>{children}</Text>;
};

export default CustomText;
