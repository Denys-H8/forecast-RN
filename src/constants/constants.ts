import { getStatusBarHeight } from 'react-native-status-bar-height';

export const MAIN_COLOR = '#fff';
export const SUB_COLOR = '#8B8EA5';
export const BG_COLOR = '#191C39';
export const ACTIVE_TEXT_COLOR = '#C5BBF6';
export const BAR_BG_COLOR = 'rgba(0, 0, 0, 0.2)';
export const BTN_BG_COLOR = 'rgba(25, 28, 57, 0.95)';
export const BTN_LOGIN_BG_COLOR = '#7B61FF';
export const STATUSBAR_HEIGHT = getStatusBarHeight();
