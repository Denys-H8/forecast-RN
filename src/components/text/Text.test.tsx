import React from 'react';
import { render } from '@testing-library/react-native';
import Text from './Text';
import constants from 'src/constants';

test('Loading snapshot', () => {
  const textTitle = render(<Text isTitle>Some Text</Text>);
  const textNotTitle = render(<Text>Some Text</Text>);
  const textWithStyle = render(<Text style={{ color: constants.SUB_COLOR }}>Some Text</Text>);

  expect(textTitle.toJSON()).toMatchSnapshot();
  expect(textNotTitle.toJSON()).toMatchSnapshot();
  expect(textWithStyle.toJSON()).toMatchSnapshot();
});
