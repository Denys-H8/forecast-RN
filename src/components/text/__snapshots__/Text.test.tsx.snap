// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "WorkSans-ExtraBold",
        "fontSize": 52,
        "lineHeight": 61,
      },
      undefined,
    ]
  }
>
  Some Text
</Text>
`;

exports[`Loading snapshot 2`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "WorkSans-Regular",
      },
      undefined,
    ]
  }
>
  Some Text
</Text>
`;

exports[`Loading snapshot 3`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "WorkSans-Regular",
      },
      Object {
        "color": "#8B8EA5",
      },
    ]
  }
>
  Some Text
</Text>
`;
